utf8parse/src/lib.rs

```diff
@@ -98,13 +98,30 @@ mod tests {
     use std::io::prelude::*;
     use std::fs::File;
 
+    // i have no idea what I'm doing here
+    impl Receiver for String {
+        fn codepoint(&mut self, c: char) {
+            self.0.print(c);
+        }
+
+        fn invalid_sequence(&mut self) {
+            self.0.print('�');
+        }
+    }
+
     #[test]
     fn utf8parse_test() {
         let mut buffer = String::new();
-        let file = File::open("UTF-8-demo.txt"); // complicated UTF-8 file
-        let file_to_string = file.read_to_string(&mut buffer);
-        let actual; // utf8parse implementation
-        let expected; // standard library implementation
+        let mut file = try!(File::open("UTF-8-demo.txt")); // complicated UTF-8 file
+        let file_to_string = try!(file.read_to_string(&mut buffer));
+        let parser = Parser::new();
+        let receiver; // ???
```

Comment:
You've got an `impl utf8parse::Receiver for String`, so the receiver would just be a `String::new()`.